components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/client/authentication/OAuthClientAuthnService.java

```diff
@@ -60,6 +75,28 @@ public List<OAuthClientAuthenticator> getClientAuthenticators() {
     public OAuthClientAuthnContext authenticateClient(HttpServletRequest request, Map<String, List> bodyContentParams) {
 
         OAuthClientAuthnContext oAuthClientAuthnContext = new OAuthClientAuthnContext();
+        try {
+            String clientId = extractClientId(request, bodyContentParams);
+            if (!StringUtils.isBlank(clientId) && Boolean.parseBoolean(isFapiApplication(clientId))) {
```

Comment:
Possible null pointer if 'isFapiApplication(clientId)' returns null

Reply:
Fixed in 2604e0460788414d7a215802ac7a3ec386fb6fda